tests/integration/cocoon/test_cocoon_portfolio_groups.py

```diff
@@ -0,0 +1,324 @@
+import unittest
+from pathlib import Path
+import pandas as pd
+import json
+import lusid
+import lusid.models as models
+from lusidtools import cocoon as cocoon
+from lusidtools.cocoon.utilities import create_scope_id
+from lusidtools import logger
+
+
+class CocoonTestPortfolioGroup(unittest.TestCase):
```

Comment:
Nice test coverage. Note that there is quite a lot of repeated code in the tests below. Consider using paramaterized or creating functions for some of the commonly used elements like you did with create_portfolio_model.

Reply:
Some gaps I did notice:

- All your test cases have portfolios in the same scope
- There is no test case for trying to update a group with a duplicate portfolio
- There is no test for trying to update a group when one of the portfolios you are trying to add does not exist
- There is no test case for trying to update a group when there are no new portfolios to add (i.e. all portfolios already added)
- There is no test case for creating a portfolio group with properties

Reply:
added these tests.

Reply:
Test Cases:

Can you please update the description of each test? Currently they are the same even for different tests.

Why in most of your tests don't you check that the created/updated portfolio groups contain the portfolios that you expect?

I think you'll find that when you update a portfolio group you will only ever add the first missing portfolio and all others will be ignored.




To do:
- _There is still no test case_ where you have portfolios in different scopes being added to a group or created alongside a group
- You have added a test for trying to update a group when one of the portfolios you are trying to add does not exist _( but need to try with one bad and one or more good to ensure that the good ones go in)_
- You have added a test case (test_10_pg_add_no_new_portfolio) for trying to update a group when there are no new portfolios to add (i.e. all portfolios already added) however this actually does add a portfolio as there are no portfolios in the group you created earlier in the test. 

Done:
- There is no test case for creating a portfolio group with properties **(Nice one)**
- There is no test case for trying to update a group with a duplicate portfolio **(Nice one)**

Reply:
Great work! Looks brilliant!